python/pyarrow/array.pxi

```diff
@@ -165,7 +170,6 @@ def array(object obj, type=None, mask=None, size=None, bint from_pandas=False,
                 from_pandas=True, safe=safe,
                 memory_pool=memory_pool)
         else:
-            import pyarrow.pandas_compat as pdcompat
```

Comment:
@wesm I'm just realising that here is one of the potential problems users may have reported about `pdcompat` import problems. This path should be supported without pandas but currently isn't.

Reply:
yes, I would agree with you. can you create a follow up issue about this? We should set up a docker-compose "no pandas" build to make sure that the project is usable without pandas

Reply:
I went ahead and opened https://issues.apache.org/jira/browse/ARROW-4640